src/formatDate.js

```diff
@@ -8,7 +8,33 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+  const deleteLastElement = fromFormatCopy.pop();
+  const partOfDate = date.split(deleteLastElement);
+  const partsOfDate = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    partsOfDate[part] = partOfDate[index];
```

Comment:
The `partOfDate` array is not correctly populated. Since `partOfDate` is derived from splitting the date string, ensure that it contains the correct segments corresponding to `fromFormatCopy`. Otherwise, `partsOfDate` will not be populated correctly.